root/static/scripts/relationship-editor/components/DialogPreview.js

```diff
@@ -97,7 +98,14 @@ const DialogPreview = (React.memo<PropsT>(({
   ) => (
     <EntityLink
       allowNew
-      content={content}
+      content={
+        nonEmpty(content)
+          ? content
+          : (
+            isDatabaseRowId(entity.id)
+              ? '' // have EntityLink determine the content
+              : (nonEmpty(entity.name) ? entity.name : l('[unknown]')))
```

Comment:
FWIW I feel this is a bit more clear
```suggestion
            isDatabaseRowId(entity.id) || nonEmpty(entity.name)
              ? '' // have EntityLink determine the content
              : l('[unknown]'))
```